### DISCUSSION TITLE
Deploying to Google Cloud Run

### DISCUSSION THREAD

**Discussion by bjayaram** [elimintz/justpy#291] [Q&A]
Really like working with JustPy, especially for SPAs.

I have everything working locally and now I want to deploy it to the cloud. I followed the instructions given here https://justpy.io/reference/justpy/ and have a Dockerfile (excerpt below) to build an image and deploy to Google Cloud Run.

> 
```
EXPOSE 80
ENV PORT 80

# Run the web service on container startup.
CMD exec uvicorn --host 0.0.0.0 --port 80 main:app
```

I have set `app = jp.app`

I am seeing this stack trace when I deploy which makes me think that I need to set `start_server` to False while calling justpy. Am I understanding this correctly?

>  
```
File "/usr/src/app/./main.py", line 115, in <module>
    jp.justpy(main, host='0.0.0.0', port='80')
  File "/usr/local/lib/python3.9/site-packages/justpy/justpy.py", line 383, in justpy
    uvicorn.run(app, host=host, port=port, log_level=UVICORN_LOGGING_LEVEL)
  File "/usr/local/lib/python3.9/site-packages/uvicorn/main.py", line 393, in run
    server.run()
  File "/usr/local/lib/python3.9/site-packages/uvicorn/server.py", line 50, in run
    loop.run_until_complete(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 618, in run_until_complete
    self._check_running()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 578, in _check_running
    raise RuntimeError('This event loop is already running')
RuntimeError: This event loop is already running
```

**Comment by elimintz** [comment]
Yes,  setting start_server to False should solve the problem. Let me know if it doesn't.

**Reply by bjayaram** [reply]
It did. Thanks.

 One thing that tripped me up is that for Google Cloud Run, when deploying a container, the default container port is 8080. I had to change it to match what I had in the Dockerfile and in my JustPy call.

**Reply by elimintz** [reply]
If it is not too much work, could you publish or send me the docker container definition file, or some version of it that just leaves in the essentials for JustPy? It would help others so could be nice to add to the project. I understand if this is something you can't do.
